### PR TITLE
Refactor references of Microsoft.Azure.Services.AppAuthentication to Azure.Identity

### PR DESCRIPTION
**Pull request by missymessa**
https://github.com/dotnet/dnceng/issues/885

### CODE REVIEW

File: src/Microsoft.DncEng.DeployServiceFabricCluster/Microsoft.DncEng.DeployServiceFabricCluster.csproj [hunk 1]
```diff
@@ -10,9 +10,9 @@
   </PropertyGroup>
 
   <ItemGroup>
+	  <PackageReference Include="Azure.Identity" />
```
Comment (garath) [inline src/Microsoft.DncEng.DeployServiceFabricCluster/Microsoft.DncEng.DeployServiceFabricCluster.csproj:13]:
```suggestion
    <PackageReference Include="Azure.Identity" />
```
Spacing is wonky.

File: src/RolloutScorer/RolloutScorerAzureFunction/RolloutScorerFunction.cs [hunk 2]
```diff
@@ -1,6 +1,5 @@
 using Azure.Identity;
 using Azure.Security.KeyVault.Secrets;
-using Microsoft.Azure.Services.AppAuthentication;
```
Comment (garath) [inline src/RolloutScorer/RolloutScorerAzureFunction/RolloutScorerFunction.cs:3]: Oh, just an unnecessary using? Nice!

Comment (missymessa) [reply src/RolloutScorer/RolloutScorerAzureFunction/RolloutScorerFunction.cs:3]: Yeah, there were a couple of those I was able to clean up :)

**Review by garath — COMMENTED** [review]
For RolloutScorere, this bit exists in deploy.yaml. The environment variable `AzureServicesAuthConnectionString` is used by `AzureServiceTokenProvider`. Ensure this value is not actually used and, if it is, reformat it in a way acceptable to `EnvironmentTokenCredential`.

https://github.com/dotnet/dnceng/blob/c55ff7197477466713293b2c40fc138023216ed2/eng/deploy.yaml#L142-L161

File: src/Microsoft.DncEng.DeployServiceFabricCluster/Helpers.cs [hunk 3]
```diff
@@ -15,20 +16,24 @@ namespace Microsoft.DncEng.DeployServiceFabricCluster;
 internal static class Helpers
 {
     private const string MsftAdTenantId = "72f988bf-86f1-41af-91ab-2d7cd011db47";
-    private static readonly AzureServiceTokenProvider TokenProvider = new AzureServiceTokenProvider();
+    private static readonly DefaultAzureCredential TokenProvider = new DefaultAzureCredential();
```
Comment (garath) [inline src/Microsoft.DncEng.DeployServiceFabricCluster/Helpers.cs:19]: This is the scary bit where the behavior differences between AzureServiceTokenProvider and DefaultAzureCredential will pop up. All downstream consumers will need to be checked for these differences (for example, if credentials are passed by environment variable).

Comment (missymessa) [reply src/Microsoft.DncEng.DeployServiceFabricCluster/Helpers.cs:19]: While refactoring, all references to the changed code was within this file. If I'm missing something, let me know.

Comment (garath) [reply src/Microsoft.DncEng.DeployServiceFabricCluster/Helpers.cs:19]: This looks fine, I'm just highlighting that this is where behavior differences are so anything that consumes this code will have to also be changed accordingly.

**Comment by missymessa** [comment]
> For RolloutScorere, this bit exists in deploy.yaml. The environment variable `AzureServicesAuthConnectionString` is used by `AzureServiceTokenProvider`. Ensure this value is not actually used and, if it is, reformat it in a way acceptable to `EnvironmentTokenCredential`.
> 
> https://github.com/dotnet/dnceng/blob/c55ff7197477466713293b2c40fc138023216ed2/eng/deploy.yaml#L142-L161

I couldn't find anywhere the `AzureServicesAuthConnectionString` was used, but the other four variables were being used, so I deleted both of them. We'll see if that breaks things when it comes time to deploy RolloutScorer.

**Comment by garath** [comment]
> I couldn't find anywhere the AzureServicesAuthConnectionString was used, but the other four variables were being used, so I deleted both of them. We'll see if that breaks things when it comes time to deploy RolloutScorer.

To be clear, AzureServicesAuthConnectionString will not be referenced in our code anywhere. It is consumed by the Microsoft.Azure.Services.AppAuthentication library. So a simple string search will return nothing. The only way to be sure is to grok how the project is expecting authentication to work.

**Comment by missymessa** [comment]
> > I couldn't find anywhere the AzureServicesAuthConnectionString was used, but the other four variables were being used, so I deleted both of them. We'll see if that breaks things when it comes time to deploy RolloutScorer.
> 
> To be clear, AzureServicesAuthConnectionString will not be referenced in our code anywhere. It is consumed by the Microsoft.Azure.Services.AppAuthentication library. So a simple string search will return nothing. The only way to be sure is to grok how the project is expecting authentication to work.

I'm going to assume that since RolloutScorer doesn't make use of the references to the `Microsoft.Azure.Services.AppAuthentication` library that was removed, that it doesn't use that parameter in the project deployment. (RolloutScorer looks like it was already updated to use `Azure.Identity`, too). 

Unless there's something automagical going on in the background for how Azure Functions are deployed?

**Comment by garath** [comment]
> To be clear, AzureServicesAuthConnectionString will not be referenced in our code anywhere. It is consumed by the Microsoft.Azure.Services.AppAuthentication library.

Oh, one idea: Write a new [TokenCredential](https://learn.microsoft.com/en-us/dotnet/api/azure.core.tokencredential?view=azure-dotnet) class that reads that environment variable and adapts it into something the new library understands.

**Comment by garath** [comment]
> I'm going to assume that since RolloutScorer doesn't make use of the references to the Microsoft.Azure.Services.AppAuthentication library that was removed, that it doesn't use that parameter in the project deployment. (RolloutScorer looks like it was already updated to use Azure.Identity, too).

Having now looked through the RolloutScorer code, I agree, that appears to be the case.

**Comment by missymessa** [comment]
> > To be clear, AzureServicesAuthConnectionString will not be referenced in our code anywhere. It is consumed by the Microsoft.Azure.Services.AppAuthentication library.
> 
> Oh, one idea: Write a new [TokenCredential](https://learn.microsoft.com/en-us/dotnet/api/azure.core.tokencredential?view=azure-dotnet) class that reads that environment variable and adapts it into something the new library understands.

I don't think this will be necessary. 

The migration guidance doc calls out how to migrate those connection strings near the beginning: https://learn.microsoft.com/en-us/dotnet/api/overview/azure/app-auth-migration?view=azure-dotnet 

Given that the migration to use `Azure.Identity` in RolloutScorer was back in [Dec 2022](https://github.com/dotnet/dnceng/commit/f343f1abf091879c5bef05a41f870668e3b771c8), and we've been able to deploy it without issues since then, it doesn't look like that variable is necessary. 

Tagging @dkurepa in case he has further insight, since he did the original migration.

**Review by garath — APPROVED** [review]

**Review by garath — APPROVED** [review]

**Comment by garath** [comment]
> Given that the migration to use Azure.Identity in RolloutScorer was back in Dec 2022, and we've been able to deploy it without issues since then, it doesn't look like that variable is necessary.

Looking through the history I suspect the change in dotnet/arcade-services#2150 did (unintentionally?) change how authentication worked. 

But that's the past.

**Comment by missymessa** [comment]
Going to merge this. @dkurepa let me know if there's anything else that should've been changed when I removed the AppAuth references.

**Pull request merged**